src/main/java/io/fabric8/elasticsearch/plugin/OpenshiftRequestContextFactory.java

```diff
@@ -130,7 +138,13 @@ private void logRequest(final RestRequest request, final UserProjectCache cache)
     private Set<String> listProjectsFor(final String user, final String token) throws Exception {
         Set<String> names = new HashSet<>();
         try {
-            ConfigBuilder builder = new ConfigBuilder().withOauthToken(token);
+            ConfigBuilder builder = new ConfigBuilder(false).withOauthToken(token)
+                                                            .withMasterUrl(openshiftMasterUrl)
+                                                            .withTrustCerts(openshiftMasterUrlTrustedCert);
+            if (openshiftCaPath != null) {
+                builder.withCaCertFile(openshiftCaPath);
+            }
+
```

Comment:
Are there additional types of Exceptions which can be thrown and need to be handled? e.g. if the user specifies an incorrect `openshiftCaPath` will the api throw a `FileNotFound` or `PermissionDenied` or some other type of exception?  I want to make sure e.g. if the user made a typo they can easily identify the problem, or if the user did not set the right file permission, etc.

Reply:
If the CA path is incorrect the following exception is thrown : 
```
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:57)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:137)
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:41)
	at io.fabric8.openshift.client.DefaultOpenShiftClient.<init>(DefaultOpenShiftClient.java:174)
	at io.fabric8.openshift.client.DefaultOpenShiftClient.<init>(DefaultOpenShiftClient.java:170)
	at io.fabric8.elasticsearch.plugin.OpenshiftClientFactory.create(OpenshiftClientFactory.java:50)
[....]
Caused by: java.io.FileNotFoundException: /tmp/junit2122595593726896985/ca.crt.does_not_exist (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at io.fabric8.kubernetes.client.internal.CertUtils.getInputStreamFromDataOrFile(CertUtils.java:55)
	at io.fabric8.kubernetes.client.internal.CertUtils.createTrustStore(CertUtils.java:61)
	at io.fabric8.kubernetes.client.internal.SSLUtils.trustManagers(SSLUtils.java:113)
	at io.fabric8.kubernetes.client.internal.SSLUtils.trustManagers(SSLUtils.java:107)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:68)
	... 29 more
```

In case of bad permissions : 

```
io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
	at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:57)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:137)
	at io.fabric8.kubernetes.client.BaseClient.<init>(BaseClient.java:41)
	at io.fabric8.openshift.client.DefaultOpenShiftClient.<init>(DefaultOpenShiftClient.java:174)
	at io.fabric8.openshift.client.DefaultOpenShiftClient.<init>(DefaultOpenShiftClient.java:170)
	at io.fabric8.elasticsearch.plugin.OpenshiftClientFactory.create(OpenshiftClientFactory.java:50)
[...]
Caused by: java.io.FileNotFoundException: /tmp/you_cant_read_it.crt (Permission denied)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at io.fabric8.kubernetes.client.internal.CertUtils.getInputStreamFromDataOrFile(CertUtils.java:55)
	at io.fabric8.kubernetes.client.internal.CertUtils.createTrustStore(CertUtils.java:61)
	at io.fabric8.kubernetes.client.internal.SSLUtils.trustManagers(SSLUtils.java:113)
	at io.fabric8.kubernetes.client.internal.SSLUtils.trustManagers(SSLUtils.java:107)
	at io.fabric8.kubernetes.client.utils.HttpClientUtils.createHttpClient(HttpClientUtils.java:68)
	... 29 more
```